### DISCUSSION TITLE
possibility of using both SPI1 and DAC

### DISCUSSION THREAD

**Discussion by flute2k3** [CommunityGD32Cores/GD32Core-New#101] [Q&A] [answered]
The bluepill plus board (https://github.com/WeActStudio/BluePill-Plus/blob/master/HDK/BluePillPlus_V10_SchDoc.pdf) has a soderable flash using SPI1 pin A4, A5, A6 & A7, while the DAC applies to PA4 and PA5 only. 

my question is:

* is it possible to use them at the same time? what about I only use DAC at pin A4 ? understood CS (A4) is not a must for SPI.
* or if not possible, can we use and then release the pins?

**Comment by maxgerhardt** [comment, marked as answer]
Well even if you were to switch between using PA4 as a DAC or SPI CS, the connected flash chip would still see a voltage on its CS pin, and if it's low enough the flash chip becomes selected and starts to try and clock in data from SCLK (PA5, the other DAC pin incidentally) and MOSI (PA7). So I don't think this would work at all and would trigger random things to happen in the flash chip with SCLK and CS having weird analog voltages on it.

So even though the SPI pins are re-mappable to different pins, this wouldn't help you since you can't change the hardware if you want to keep using that bluepill board. It would then just not put the SPI signals on the right pins. And sadly, the DAC pins are not remappable.

This leaves little options but to cut traces to the flash chip and resolder them to the alternative SPI0 pins, or SPI1 pins. 

![grafik](https://github.com/CommunityGD32Cores/ArduinoCore-GD32/assets/26485477/0ff272da-d577-4284-b6d4-b5f120b3035a)

**Comment by flute2k3** [comment]
well I now think this is a bug of bluepill plus design, they should use SPI2 as you pointed out.
